OpenEdXMobile/src/main/java/org/edx/mobile/inapppurchases/BillingProcessor.kt

```diff
@@ -165,44 +183,51 @@ class BillingProcessor @Inject constructor(@ApplicationContext val context: Cont
      */
     private fun retryBillingServiceConnectionWithExponentialBackoff() {
         handler.postDelayed(
-            { billingClient.startConnection(this@BillingProcessor) },
+            { billingClient.startConnection(billingClientStateListener) },
             RECONNECT_TIMER_START_MILLISECONDS
         )
     }
 
     /**
-     * Calls the billing client functions to query sku details for in-app SKUs. SKU details are
-     * useful for displaying item names and price lists to the user, and are required to make a
-     * purchase.
+     * Calls the billing client functions to query product details for in-app products. Product
+     * details are useful for displaying item names and price lists to the user, and are required to
+     * make purchase.
      *
-     * @param productId SKU of the product
-     * @param listener [SkuDetailsResponseListener]
+     * @param productId Id of the product
+     * @return Details of the product
      * */
-    fun querySyncDetails(productId: String, listener: SkuDetailsResponseListener) {
-        startConnection()
-        billingClient.querySkuDetailsAsync(
-            SkuDetailsParams.newBuilder()
-                .setType(BillingClient.SkuType.INAPP)
-                .setSkusList(listOf(productId))
-                .build(), listener
+    suspend fun querySyncDetails(productId: String): ProductDetailsResult {
+        connectIfNeeded()
+        val productList = listOf(
+            QueryProductDetailsParams.Product.newBuilder()
+                .setProductId(productId)
+                .setProductType(BillingClient.ProductType.INAPP)
+                .build()
         )
+
+        return withContext(Dispatchers.IO) {
+            billingClient.queryProductDetails(
+                QueryProductDetailsParams
+                    .newBuilder()
+                    .setProductList(productList)
+                    .build()
+            )
+        }
     }
 
     /**
      * Method to query the Purchases async and returns purchases details for currently owned items
      * bought within the app.
      *
-     * @param listener callback interface to get the Purchases.
-     * */
-    fun queryPurchase(listener: PurchasesResponseListener) {
-        startConnection()
-        billingClient.queryPurchasesAsync(
-            BillingClient.SkuType.INAPP, listener
-        )
-    }
-
-    fun isConnected(): Boolean {
-        return billingClient.isReady
+     * @return List of purchases
+     **/
+    suspend fun queryPurchases(): List<Purchase> {
+        connectIfNeeded()
+        return billingClient.queryPurchasesAsync(
```

Comment:
As per doc [querPurchasesAsync](https://developer.android.com/reference/com/android/billingclient/api/BillingClient#queryPurchasesAsync(com.android.billingclient.api.QueryPurchasesParams,%20com.android.billingclient.api.PurchasesResponseListener)) requires two params.

Reply:
This one is default extension function provided by the library that provides result as a return type instead of using a callback.